### ISSUE TITLE
Control panel - GEDCOM tags

### ISSUE THREAD

**Issue by webtrees-pesz** [fisharebest/webtrees#4501]
Currently, only level 1 tags are supported via Control panel - GEDCOM tags. It would also be useful to support subtags on level 2. Example: MARR.TYPE

There should be the possibility to control all 'subtags‘ of events and facts in Control panel - GEDCOM tags

**Comment by fisharebest** [comment]
> There should be the possibility to control all 'subtags‘ of events and facts in Control panel - GEDCOM tags

There are probably a thousand separate tag/subtag/... combinations.
It is not practical or desirable to make all of them configurable.

> Example: MARR.TYPE

This has been removed from the list of tags that are "hidden by default".

**Comment by ric2016** [comment]
I agree that everything should be configurable somehow, regardless of the number of options.

Is it desirable to have a large number of forum threads instead for all kinds of special cases, leading to solutions that still do not work for all users?

**Comment by fisharebest** [comment]
> I agree that everything should be configurable somehow, regardless of the number of options.

There are approx 50 facts/events for FAM/INDI

Each of these can have TYPE/DATE/PLAC/CAUS/RELI/AGNC/ADDR/NOTE/SOUR/OBJE/RESN...

That's 550 configuration options.

There are about 50 tags under NAME.

Add in level 3 tags, and other object types, and we will easily have over 1000.

Do you really want a form with 1000+ options, most of which will never be used?

**Comment by ric2016** [comment]
There are different ways to make this manageable. It doesn't have to be all in a single form.

One possibility is to use separate forms, one per event. Mabe even add a configuration option (for admins only) to the edit dialogs themselves: So when I edit an event and miss a tag, or see a tag that is never used, I can change this directly for this type of event, within the event structure as presented (when editing with all gedcom tags, they all show up already anyway).

Another approach is to start with the tag, e.g. AGNC, and configure for which specific events to show it.

**Comment by webtrees-pesz** [comment]
Greg,
the idea was to mark infrequently used tags and their exceptions in a "structure oriented" list for the different levels.

See attached examples for 5.5.1 and 7.0: 
[Lineage-Linked-Structure for 5.5.1 and 7.0.xlsx](https://github.com/fisharebest/webtrees/files/9042070/Lineage-Linked-Structure.for.5.5.1.and.7.0.xlsx)


The same method can be used to build different forms for different GEDCOM versions.

**Comment by ric2016** [comment]
[See this forum thread](https://www.webtrees.net/index.php/en/forum/4-customising/37154-customization-of-gedcom-tag-visibility) for an example of how this could be handled.
